spec/features/index_link_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe "a client viewing the home page", type: :feature do
+
+  let!(:user) do
+    User.create(
+      username: 'username123',
+      email: 'email@test.com',
+      password: 'password123',
+      password_confirmation: 'password123'
+      )
+  end
+
+  let!(:link) do
+    Link.create(
+      title: 'Test Link',
+      url: 'http://www.testlink.com/',
+      user: user
+    )
+  end
+
+  let!(:link2) do
+    Link.create(
+      title: 'Test Link2',
+      url: 'http://www.testlink2.com/',
+      user: user
+    )
+  end
+
+  context "with links present in database" do
+
+    it "should see links visible on the page" do
+      visit root_path
+      expect(page.find("#link-#{link.id}").present?).to eq true
+      expect(page.find("#link-#{link2.id}").present?).to eq true
```

Comment:
We generally try to shy away from referencing IDs or CSS classes in our assertions.  We only ever really use them if we don't have any label/piece of text to hinge off of, or when we're using `within`.  Try to assert against things a user would see on the page without having to open up dev tools, so in this case.. it's probably the titles of the links, "Test Link" and "Test Link2".